app/models/concerns/foreman_teamdynamix/host_managed_extensions.rb

```diff
@@ -0,0 +1,21 @@
+module ForemanTeamdynamix
+  module HostManagedExtensions
+    extend ActiveSupport::Concern
+
+    def td_api
+      @td_api || TeamdynamixApi.new
+    end
+
+    included do
+      after_create :create_teamdynamix_asset
```

Comment:
use ```before_validation :create_teamdynamix_asset, on: :create```

This way you don't have to worry about saving again below and turning off validation etc...

Reply:
good idea. 
save without validation bypass was a leftover statement from the merge conflict.